src/tree.rs

```diff
@@ -331,24 +331,15 @@ impl<T> Tree<T> {
             return Result::Err(error.expect("Tree::remove_node_lift_children: Missing an error value on finding an invalid NodeId."));
         }
 
-        if !self.node_has_parent(&node_id) {
-            self.clear_parent_of_children(&node_id);
-        } else {
-            let parent_id = self.get_unsafe(&node_id)
-                .parent()
-                .expect("Tree::remove_node_lift_children: node_has_parent() is true, but unwrapping said parent failed.")
-                .clone();
-
-            let children = self.get_unsafe(&node_id)
-                .children()
-                .clone();
-
-            for child_id in children {
+        if let Some(parent_id) = self.get_unsafe(&node_id).parent().cloned() {
+            //attach children to parent
+            for child_id in self.get_unsafe(&node_id).children().clone() {
                 self.set_as_parent_and_child(&parent_id, &child_id);
             }
+        } else {
+            self.clear_parent_of_children(&node_id);
```

Comment:
Is this the best way to handle this?

I think using `remove_node_lift_children` on the root should be an `Error`, because there is no way to lift them in that case. Making them orphans is unexpected, don't you think so?